docs/content/examples/tutorials/mailbox-compression.md

```diff
@@ -0,0 +1,142 @@
+---
+title: 'Tutorials | Mailbox Compression'
+---
+
+## Overview
+
+Compression allows all messages compressed or not to be READ, and/or compressed on save, regardless of the Algorithm chosen for write. Dovecot message compression plugin supports a variety of algorithms, please review them and the options available at:
+
+* Dovecot 2.3- / DMS 15-: [dovecot-zlib](https://doc.dovecot.org/2.3/configuration_manual/zlib_plugin/)
+* Dovecot 2.4+ / DMS 16+: [mail-compress](https://doc.dovecot.org/2.4.1/core/plugins/mail_compress.html)
+
+Using `zstd` method seems to be the best and faster algorithm to date, according to a recent [mail_compress-benchmark](https://github.com/dovecot/documentation/edit/main/docs/core/plugins/mail_compress.md).
+
+While compression can be memory intensive for large messages, you can configure the plugin to limit the amount of memory consumed by the workers. Examples shown mention the `vsz_limit` setting to limit to 1GB RAM for instance.
+
+## Setup
+
+### Dovecot < 2.4 / DMS < 16
+
+To configure  as a dovecot plugin, create a file at `docker-data/dms/config/dovecot/12-compress-2.3.conf` and place the following in it:
+
+```
+mail_plugins = $mail_plugins zlib
+
+# Enable WRITE compression on saving messages:
+plugin {
+  # zlib_save = gz
+  # zlib_save_level = 1
+  zlib_save = zstd
+}
+
+# Enable READ compression for IMAP:
+# Enable Zlib for imap
+protocol imap {
+  mail_plugins = $mail_plugins zlib
+}
+
+# Enable READ compression for POP3:
+protocol pop3 {
+  mail_plugins = $mail_plugins zlib
+}
+
+# Increase memory allowed for imap as it costs more to read compressed files
+service imap {
+  vsz_limit = 1GB
+}
+```
+
+    Adjust the settings to tune for your desired memory limits and CPU usage.
+
+    The following compression levels are supported for `zlib_save_level`:
+    
+    | Name   |   Minimum          |   Default | Maximum |
+    |--------|--------------------|-----------|-------- |
+    | `bz2`  | 1                  | 9         | 9       |
+    | `gz`   | 0 (no compression) | 6         | 9       |
+    | `lz4`  | 1                  | 1         | 9       |
+    | `zstd` | 1                  | 3         | 22      |
+
+
+### Dovecot 2.4+ / DMS 16+
+
+From Dovecot 2.4, the compression plugin name and setup have changed: [mail-compress](https://doc.dovecot.org/2.4.1/core/plugins/mail_compress.html). Create a file at `docker-data/dms/config/dovecot/12-compress.conf` and place the following in it:
+
+```
+# Enable compression plugin globally for reading/writing:
+mail_plugins {
+  mail_compress = yes
+}
+
+# Enable WRITE compression on saving messages:
+mail_compress_write_method = zstd
+
+# Increase memory allowed for imap as it costs more to read compressed files
+service imap {
+  vsz_limit = 1GB
+}
```

Comment:
This is only considering imap, as noted in other feedback that may be inaccurate. If memory is an issue it could be relevant to consider the cost for compression for inbound mail large enough to warrant this, in addition to the outbound read to clients over IMAP and potentially POP3.

A global upgrade to that setting may be better?